wandb/sdk/data_types.py

```diff
@@ -2229,6 +2229,8 @@ def val_to_json(
     typename = util.get_full_typename(val)
 
     if util.is_pandas_data_frame(val):
+        assert run
+        run.history._data = dict()
```

Comment:
I think this is problematic.  Users can call `wandb.log` in multiple locations for the same "step" before flushing it.  i.e.

```
wandb.log({"metric_1": N}, commit=False)
wandb.log({"dataframe": df})
```

The above should log both the dataframe and "metric_1".  I believe your code will wipe out "metric_1".  In fact, depending on the keys that are in your log call and the order they happen to be iterated in we drop other keys in the same step.  I'm not sure how to best solve this, even if we only wiped out existing tables in the current histories data we could drop data...

Reply:
good point, should we just automatically convert to a wandb table when someone logs a pandas dataframe ?

Reply:
I guess for now we can just remove data frames values.  Bonus would be to automatically create wandb tables which seems sane and should fix all cases.

Reply:
i implemented the second approach but I think it will only work in online mode, since `wandb.Tables` are logged as artifacts. Is this correct and, if so, a problem ?